crates/node-api/src/engine/traits.rs

```diff
@@ -2,12 +2,38 @@ use crate::{validate_version_specific_fields, AttributesValidationError, EngineA
 use reth_primitives::{
     revm::config::revm_spec_by_timestamp_after_merge,
     revm_primitives::{BlobExcessGasAndPrice, BlockEnv, CfgEnv, SpecId},
-    Address, ChainSpec, Header, B256, U256,
+    Address, ChainSpec, Header, SealedBlock, B256, U256,
 };
-use reth_rpc_types::engine::{
-    OptimismPayloadAttributes, PayloadAttributes as EthPayloadAttributes, PayloadId, Withdrawal,
+use reth_rpc_types::{
+    engine::{
+        ExecutionPayloadEnvelopeV2, ExecutionPayloadEnvelopeV3, OptimismPayloadAttributes,
+        PayloadAttributes as EthPayloadAttributes, PayloadId, Withdrawal,
+    },
+    ExecutionPayloadV1,
 };
 
+/// Represents a built payload type that contains a built [SealedBlock] and can be converted into
+/// engine API execution payloads.
+pub trait BuiltPayload: Send + Sync + std::fmt::Debug {
+    /// Initializes the payload with the given initial block.
+    fn new(id: PayloadId, block: SealedBlock, fees: U256) -> Self;
+
+    /// Returns the built block (sealed)
+    fn block(&self) -> &SealedBlock;
+
+    /// Returns the fees collected for the built block
+    fn fees(&self) -> U256;
+
+    /// Converts the type into the response expected by `engine_getPayloadV1`
+    fn into_v1_payload(self) -> ExecutionPayloadV1;
+
+    /// Converts the type into the response expected by `engine_getPayloadV2`
+    fn into_v2_payload(self) -> ExecutionPayloadEnvelopeV2;
+
+    /// Converts the type into the response expected by `engine_getPayloadV2`
+    fn into_v3_payload(self) -> ExecutionPayloadEnvelopeV3;
```

Comment:
these are fine for now, but perhaps we want to reconsider this eventually and instead just require that the type is Serialize or can be converted into serde_Value